### PR TITLE
Allow customizing securityContext, resources, etc. for Fulcio

### PR DESCRIPTION
**Pull request by k4leung4**
Signed-off-by: Kenny Leung <kleung@chainguard.dev>

#### Summary

    - allow specifying multiple container ports
    - allow specifying multiple hosts/tls for ingress
    - allow specifying ingress backend service name
    - allow specifying deployment name
    - allow specifying securityContext for create*jobs and deployment
    - allow specifying resources for deployment
    - allow specifying annotations for pods, for prometheus scrapping
    - change json config to use toPrettyJson for readability

#### Ticket Link

Fixes

#### Release Note

```release-note

```

### CODE REVIEW

File: charts/fulcio/templates/_helpers.tpl [hunk 1]
```diff
@@ -152,7 +162,7 @@ Return the appropriate apiVersion for ingress.
 Print "true" if the API pathType field is supported
 */}}
 {{- define "ingress.supportsPathType" -}}
-{{- if (semverCompare "<1.18-0" .Capabilities.KubeVersion.Version) -}}
+{{- if (semverCompare "<1.18-0" $.Capabilities.KubeVersion.Version) -}}
```
Comment (hectorj2f) [inline charts/fulcio/templates/_helpers.tpl:165]: typo: $.Capabilities

File: charts/fulcio/README.md [hunk 2]
```diff
@@ -69,8 +69,13 @@ The following table lists the configurable parameters of the Fulcio chart and th
...
 | server.ingress.annotations | object | `{}` |  |
 | server.ingress.enabled | bool | `true` |  |
-| server.ingress.path | string | `"/"` |  |
-| server.ingress.tls | object | `{}` |  |
+| server.ingress.hosts | array | [] |  |
+| server.ingress.hosts.host | string |  |  |
+| server.ingress.hosts.path | string | `"/"` |  |
+| server.ingress.hosts.service_name | string |  |  |
+| server.ingress.tls | array | `[]` |  |
+| server.ingress.tls.hosts | array | `[]` |  |
```
Comment (hectorj2f) [inline charts/fulcio/README.md:77]: I think we are missing the tlsEnabled property here.

File: charts/fulcio/values.schema.json [hunk 3]
```diff
@@ -64,8 +64,14 @@
...
-                    "path": "/",
-                    "tls": {}
+                    "hosts": [
+                      {
+                        "host": "fulcio.localhost",
+                        "path": "/"
+                      }
+                    ],
+                    "tls": [],
+                    "tlsEnabled": false
```
Comment (hectorj2f) [inline charts/fulcio/values.schema.json:74]: Question: why do we need this flag? Wouldn’t it be enough with the array size as a flag?

File: charts/fulcio/templates/_helpers.tpl (hunk 1 above)
Comment (k4leung4) [reply charts/fulcio/templates/_helpers.tpl]: thanks. fixed.

File: charts/fulcio/README.md (hunk 2 above)
Comment (k4leung4) [reply charts/fulcio/README.md:77]: added, fixed.

File: charts/fulcio/values.schema.json (hunk 3 above)
Comment (k4leung4) [reply charts/fulcio/values.schema.json]: good point. removed.

**Review by cpanato — APPROVED** [review]
thanks

i will defer to @hectorj2f

**Review by hectorj2f — APPROVED** [review]

**Pull request merged**